modeling/modeling.py

```diff
@@ -0,0 +1,204 @@
+import pandas as pd
+import numpy as np
+import logging
+from typing import Tuple
+
+# evaluation imports
+from sklearn.metrics import f1_score, recall_score, precision_score
+from sklearn.model_selection import GridSearchCV
+
+# one-million-post utils
+from utils import loading, feature_engineering
+
+# mlflow
+import mlflow
+from mlflow.sklearn import save_model
+from modeling.config import TRACKING_URI, EXPERIMENT_NAME
+
+# set logging
+logger = logging.getLogger(__name__)
+logging.basicConfig(format="%(asctime)s: %(message)s")
+logging.getLogger("pyhive").setLevel(logging.CRITICAL)  # avoid excessive logs
+logger.setLevel(logging.INFO)
+
+
+
+def compute_and_log_metrics(
+    y_true: pd.Series, y_pred: pd.Series, split: str="train"
+) -> Tuple[float, float, float]:
+    """Computes and logs metrics to mlflow and logger
+
+    Args:
+        y_true: The true target classification
+        y_pred: The predicted target classification
+        split: The split of the dataset ["test", "val", "train"]
+
+    Returns:
+        f1: The f1_score
+        precision: The precision_score
+        recall: The recall_score
```

Comment:
I think it would be beneficial to print and return the confusion matrix

Reply:
done (but needs to be integrated with utils.scoring)